cernopendata/modules/theme/assets/semantic-ui/js/search/components/RangeAggregation.js

```diff
@@ -0,0 +1,206 @@
+import _debounce from 'lodash/debounce';
+import PropTypes from "prop-types";
+import React, { useState, useMemo, useEffect, useRef, useCallback } from "react";
+import { useSelector } from "react-redux";
+import { FlexibleWidthXYPlot, VerticalRectSeries, Hint } from "react-vis";
+import { Card } from "semantic-ui-react";
+import RangeSlider from './RangeSlider';
+import { withState } from 'react-searchkit';
+import { safeParse } from "../utils";
+
+const HALF_BAR_WIDTH = 0.4;
+const SELECTED_COLOR = "#91d5ff";
+const DESELECTED_COLOR = "#bfbfbf";
+const HIGHLIGHT_COLOR = "#69c0ff";
+
+
+function extractBuckets(resultsAggregations, key) {
+    if (resultsAggregations && resultsAggregations[key] && Array.isArray(resultsAggregations[key].buckets)) {
+        return resultsAggregations[key].buckets;
+    }
+    return [];
+}
+
+function getInitialHistogramData(initialBuckets) {
+  const data = initialBuckets.map(item => {
+    const key = Number(item.key_as_string); 
+    return {
+      x0: key - HALF_BAR_WIDTH,
+      x: key + HALF_BAR_WIDTH,
+      y: item.doc_count,
+      color: DESELECTED_COLOR,
+    };
+  });
+  return data;
+}
+
+function getHistogramData(initialData, [lower, upper]) {
+  const lowerNum = safeParse(lower, 1900);
+  const upperNum = safeParse(upper, 2025);
+  const data = initialData.map((item, index) => {
+    const { x0, x, y } = item;
+    const bucketKey = x - HALF_BAR_WIDTH;
+    const color = (bucketKey >= lowerNum && bucketKey <= upperNum)
+      ? SELECTED_COLOR
+      : DESELECTED_COLOR;
+    return {
+      x0,
+      x,
+      y,
+      index,
+      color,
+    };
+  });
+  return data;
+}
+
+const RangeAggregation = (props) => {
+  const { title, agg, currentQueryState } = props;
+  const resultsAggregations = useSelector(
+    (state) => state.results.data.aggregations
+  );
+  const initialBuckets = extractBuckets(resultsAggregations, agg.aggName)
+  const keys = initialBuckets.map(bucket => Number(bucket.key_as_string));
+  const min = Math.min(...keys) || 1900;
+  const max = Math.max(...keys) || 2025;
```

Comment:
this should be configurable otherwise next year will be not taken into account....maybe you can do something like  pseudocode: `date.currentYear()`